### PR TITLE
Prototype a cohort level MAF analysis tool card

### PR DESCRIPTION
**Pull request by siosonel**
## Description
Created this card for the April release. Will convert this draft PR to `ready` once an icon is available for the MAF card.

## Checklist

- [x] Added proper unit tests
- [x] Left proper TODO messages for any remaining tasks
- [x] Scanned for web accessibility with **aXe**, and mitigated or documented
      flagged issues

### CODE REVIEW

File: packages/portal-proto/src/features/user-flow/workflow/registeredApps.tsx [hunk 1]
```diff
@@ -228,6 +228,18 @@ export const REGISTERED_APPS = [
     noDataTooltip:
       "Current cohort does not have gene expression data available for visualization.",
   },
+  {
+    name: "Cohort Level MAF",
+    // TODO: replace with a MAF icon
+    icon: <BAMSlicingDownloadIcon aria-hidden="true" />,
```
Comment (siosonel) [inline packages/portal-proto/src/features/user-flow/workflow/registeredApps.tsx]: @wteouchicago Please generate an icon for this card. For now, I've used another icon only to prototype.

File: packages/portal-proto/src/features/user-flow/workflow/registeredApps.tsx [hunk 2]
```diff
@@ -228,6 +229,18 @@ export const REGISTERED_APPS = [
...
       "Current cohort does not have gene expression data available for visualization.",
   },
+  {
+    name: "Cohort Level MAF",
+    // TODO: replace with a MAF icon
+    icon: <CohortLevelMAFIcon aria-hidden="true" />,
+    tags: ["ssm"],
+    hasDemo: false,
+    countsField: "sequenceReadCaseCount",
+    description: "Combine and download cohort level MAF data",
```
Comment (wteouchicago) [inline packages/portal-proto/src/features/user-flow/workflow/registeredApps.tsx]: extreme nitpick on my end. can you add a period "." at the end of the description (for consistency with the other descriptions)?

**Comment by wteouchicago** [comment]
verified the card is showing up with the icon. 👍

**Comment by siosonel** [comment]
@craigrbarnes I also need to change the `countsField: "sequenceReadCaseCount" - is there one that I can use that is specific to MAF files?

File: packages/portal-proto/src/features/user-flow/workflow/registeredApps.tsx (hunk 2 above)
Comment (siosonel) [reply packages/portal-proto/src/features/user-flow/workflow/registeredApps.tsx]: I'll add the period in my next commit when the `countsField` is changed to something more appropriate.

**Comment by wteouchicago** [comment]
Hi @siosonel 

> @craigrbarnes I also need to change the `countsField: "sequenceReadCaseCount" - is there one that I can use that is specific to MAF files?

We are looking into the best way for doing this. Quick question: can you give us a bit more details on what data the tool uses? I assumed just MAF files, but want to make sure. :)

**Comment by siosonel** [comment]
> bit more details on what data the tool

per Xin:

```js
const filters = {
		op: 'and',
		content: [
			{ op: '=', content: { field: 'data_format', value: 'MAF' } },
			{ op: '=', content: { field: 'experimental_strategy', value: 'WXS' or 'Targeted Sequencing' } }, 
			{ op: '=', content: { field: 'analysis.workflow_type', value: 'Aliquot Ensemble Somatic Variant Merging and Masking' } }, 
			{ op: '=', content: { field: 'access', value: 'open' } } 
		]
	}
```

**Comment by craigrbarnes** [comment]
I created a branch from this with the MAF  filter query added to the `countsQuery`. If the counts for the MAF cards work as expected, then merge it to this branch. 

Note that we are planning a change in how each card computes counts and will inform you and document the change when it's ready.

**Comment by siosonel** [comment]
@craigrbarnes EDIT: I am now seeing case counts in Chrome after restarting `npm run dev`, but  `Safari` in incognito mode does not load the case count at all, the request to `/cohorts` has  the `Context ID not provided` error).


Screenshot 2024-03-21 at 3 37 19 PM


(Outdated comment below)
I tried the `add_maf_count` branch, but it gives `0`. Xin's query uses the `/files` REST API endpoint, not `/cases`. 

Screenshot 2024-03-21 at 3 00 20 PM

**Comment by siosonel** [comment]
@craigrbarnes Please see my updated comment above, I am now able to see case counts in the MAF card. I merged your branch to this branch. 

Also, just a note that the app displays the number of files, while the card displays the number of cases, so the counts in the tool versus the card are not expected to match up, in case anyone notices.

Screenshot 2024-03-21 at 3 44 11 PM

Screenshot 2024-03-21 at 3 45 14 PM

**Comment by craigrbarnes** [comment]
Are still encountering issues with the counts? If you happy with the state of this PR I will merge it.

Regarding the files vs. cases mismatch, the case counts on the app cards prevent an app from running if it does not have data. As long as using MAF case counts does not prevent the app from launching when there are the appropriate files in the cohort, the mismatch should be acceptable.  We do want to ensure that the app always is selectable if it has the appropriate data.

Note that the planned change to App card counts will allow you to use any endpoint to get counts, as each card will need to use or implement a `getCounts` hook.

**Comment by siosonel** [comment]
> Are still encountering issues with the counts? If you happy with the state of this PR I will merge it.

Nope, everything looks good, thanks!

**Comment by wteouchicago** [comment]
Hi @siosonel -- I look a look with Safari (incognito mode) and I was able to see the case counts on the MAF card. Just FYI.
Screenshot 2024-03-22 at 10 11 04 AM

**Review by craigrbarnes — APPROVED** [review]

**Review by craigrbarnes — CHANGES_REQUESTED** [review]
I reran the pipeline last night and I see it failed with a build error:
```
src/features/proteinpaint/CohortLevelMafWrapper.tsx(89,5): error TS2322: Type '{ host: string; genome: string; filter0: any; launchGdcMaf: true; }' is not assignable to type 'MafUiArg'.
  Object literal may only specify known properties, and 'genome' does not exist in type 'MafUiArg'.
npm ERR! Lifecycle script `compile` failed with error: 
npm ERR! Error: command failed 
npm ERR!   in workspace: portal-proto@2.12.0 
npm ERR!   at location: /builds/8Sbmo3DL/4/nci-gdc/front-end/gdc-frontend-framework/packages/portal-proto 
```

File: packages/portal-proto/src/features/proteinpaint/CohortLevelMafWrapper.tsx [hunk 3]
```diff
@@ -75,6 +75,7 @@ interface MafUiArg {
   nobox?: boolean;
   hide_dsHandles?: boolean;
   launchGdcMaf: true;
+  genome?: string;
```
Comment (siosonel) [inline packages/portal-proto/src/features/proteinpaint/CohortLevelMafWrapper.tsx:78]: @craigrbarnes this should fix the build issue. Also, I [created a temporary PR (now closed) ](https://github.com/NCI-GDC/gdc-frontend-framework/pull/1039) that caused the gitlab check to rerun and report a passing check for this PR (the two branches have the same commit history).

**Review by craigrbarnes — APPROVED** [review]

**Pull request merged**